base/src/main/java/io/spine/base/Environment.java

```diff
@@ -69,101 +176,57 @@ public Environment createCopy() {
     }
 
     /**
-     * Restores the state from the instance created by {@link #createCopy()}.
-     *
-     * <p>Call this method when cleaning up tests that modify {@code Environment}.
-     */
-    @VisibleForTesting
-    public void restoreFrom(Environment copy) {
-        // Make sure this matches the set of fields copied in the copy constructor.
-        this.tests = copy.tests;
-    }
-
-    /**
-     * Verifies if the code currently runs under a unit testing framework.
+     * Determines whether the current environment is the same as the specified one.
      *
-     * <p>The method returns {@code true} if the following packages are discovered
-     * in the stacktrace:
-     * <ul>
-     *     <li>{@code org.junit}
-     *     <li>{@code org.testng}
-     * </ul>
+     * <p>If {@linkplain #register(EnvironmentType) custom env types have been defined},
+     * goes through them in the latest-registered to earliest-registered order.
+     * Then, checks {@link Tests} and {@link Production}.
      *
-     * @return {@code true} if the code runs under a testing framework, {@code false} otherwise
+     * @return the current environment type.
      */
-    @SuppressWarnings({
-            "DynamicRegexReplaceableByCompiledPattern", // OK as we cache the result
-            "DuplicateStringLiteralInspection" // used in another context
-    })
-    public boolean isTests() {
-        // If we cached the value before, return it.
-        if (tests != null) {
-            return tests;
-        }
-
-        // Check the environment variable. We may run under unknown testing framework or
-        // tests may require production-like mode, which they simulate by setting
-        // the property to `false`.
-        String testProp = System.getProperty(ENV_KEY_TESTS);
-        if (testProp != null) {
-            testProp = testProp.replaceAll("\"' ", "");
-            this.tests = (String.valueOf(true)
-                                .equalsIgnoreCase(testProp)
-                    || "1".equals(testProp));
-            return this.tests;
-        }
-
-        // Check stacktrace for known frameworks.
-        String stacktrace = Throwables.getStackTraceAsString(new RuntimeException(""));
-        if (stacktrace.contains("org.junit")
-                || stacktrace.contains("org.testng")) {
-            this.tests = true;
-            return true;
+    public boolean is(EnvironmentType type) {
```

Comment:
Let's bring back `isTests()` and `isProduction()` and mark them `@Deprecated` in favor of `... .enabled()`.
It's a proper way of propagating the public API changes.